source/puppet/4.1/config_ssl_external_ca.markdown

```diff
@@ -201,230 +107,70 @@ Credential                        | File location
+Certificate     | Purpose                     | File Location
+----------------|--------------------------------------------
+ca_crt.pem      | Intermediate CA Certificate | /etc/puppetlabs/puppet/ssl/ca/ca_crt.pem
+ca_key.pem      | Intermediate CA Key         | /etc/puppetlabs/puppet/ssl/ca/ca_key.pem
+root_crt.pem    | Root CA Certificate         | /etc/puppetlabs/puppet/ssl/ca/root_crt.pem
 
-With these files in place, the web server should be configured to:
+> Note: The root_crt.pem can actually be placed anywhere, however this doc assumes you placed it as shown
 
-* Use the Agent+Root CA bundle, the master's certificate, and the master's key.
-* Set the verification header (as specified in the master's [`ssl_client_verify_header` setting][verify_header]).
-* Set the client DN header (as specified in the master's [`ssl_client_header` setting][client_header]).
+> Note: The ca_key.pem needs to have any passphrase removed to match the expectations of the Puppet CA
 
-An example of this configuration for Apache:
+All of the files placed should have owner set to `puppet:puppet` and permissions of `0600`.
 
-~~~ apache
-Listen 8140
-<VirtualHost *:8140>
-    SSLEngine on
-    SSLProtocol ALL -SSLv2
-    SSLCipherSuite ALL:!ADH:RC4+RSA:+HIGH:+MEDIUM:-LOW:-SSLv2:-EXP
+Next, you have to generate the CA bundle to be placed on the master as well as any agents you create.  This is achieved by combining the Root CA and Intermediate CA certificates into one PEM file.
 
-    # Replace with the value of `puppet master --configprint hostcert`
-    SSLCertificateFile "/path/to/master.pem"
-    # Replace with the value of `puppet master --configprint hostprivkey`
-    SSLCertificateKeyFile "/path/to/master.key"
+```
+cd /etc/puppetlabs/puppet/ssl/ca
+cat ca_cert.pem root_crt.pem > ../certs/ca.pem
+```
+> Note: You also need to install a CRL file for the CA.  If you do not have one pre-generated from the Root CA you can easily create one by first executing `puppet cert generate fakehost` and then revoking this certificate with `puppet cert clean fakehost`.  If you do have a pre-generated CRL install it into `/etc/puppetlabs/puppet/ssl/ca/ca_crl.pem.
 
-    # Replace with the value of `puppet master --configprint localcacert`
-    SSLCACertificateFile "/path/to/ca_bundle_for_master.pem"
-    SSLCertificateChainFile "/path/to/ca_bundle_for_master.pem"
+You now need to generate a new certificate for the puppet master to use.  Remember to include the `dns_alt_names` that this puppet master will need to service.
 
-    # Allow only clients with a SSL certificate issued by the intermediate CA with
-    # common name "Puppet Agent CA"  Replace "Puppet Agent CA" with the CN of your
-    # Agent CA certificate.
-    <Location />
-        SSLRequire %{SSL_CLIENT_I_DN_CN} eq "Puppet Agent CA"
-    </Location>
+```
+puppet cert generate puppetserver.my.domain.net --dns_alt_names=puppetserver,puppet
```

Comment:
@adrienthebo The puppetserver is functioning as an intermediate and is responsible for signing all agent certificates, master or otherwise.